src/main/scala/viper/gobra/translator/implementations/translator/BuiltInMembersImpl.scala

```diff
@@ -0,0 +1,454 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2011-2020 ETH Zurich.
+
+package viper.gobra.translator.implementations.translator
+
+import viper.gobra.ast.{internal => in}
+import viper.gobra.frontend.info.base.BuiltInMemberTag
+import viper.gobra.frontend.info.base.BuiltInMemberTag.{BuiltInFPredicateTag, BuiltInFunctionTag, BuiltInMPredicateTag, BuiltInMemberTag, BuiltInMethodTag, ChannelInvariantMethodTag, CloseFunctionTag, ClosedMPredTag, ClosureDebtMPredTag, CreateDebtChannelMethodTag, InitChannelMethodTag, IsChannelMPredTag, PredTrueFPredTag, RecvChannelMPredTag, RecvGivenPermMethodTag, RecvPermMethodTag, RedeemChannelMethodTag, SendChannelMPredTag, SendGotPermMethodTag, SendPermMethodTag, TokenMPredTag}
+import viper.gobra.reporting.Source
+import viper.gobra.theory.Addressability
+import viper.gobra.translator.Names
+import viper.gobra.translator.interfaces.translator.BuiltInMembers
+import viper.gobra.translator.interfaces.{Collector, Context}
+import viper.gobra.translator.util.PrimitiveGenerator
+import viper.gobra.util.Computation
+import viper.gobra.util.Violation.violation
+
+import scala.language.postfixOps
+
+
+/**
+  * Encodes built-in members by translating them to 'regular' members and calling the corresponding encoding
+  */
+class BuiltInMembersImpl extends BuiltInMembers {
```

Comment:
Seeing everything that is in here makes me doubt that this right now is the right approach. That said, we should definitely merge it for now and then after the deadline has settled down and we have time to redesign something, then we can look at it again.